src/libaktualizr/bootloader/bootloader.cc

```diff
@@ -94,3 +119,42 @@ void Bootloader::updateNotify() const {
       throw NotImplementedException();
   }
 }
+
+bool Bootloader::supportRebootDetection() const { return reboot_detect_supported_; }
+
+bool Bootloader::rebootDetected() const {
+  if (!reboot_detect_supported_) {
+    return false;
+  }
+
+  // true if set in storage and no volatile flag
+
+  bool beacon_exists = boost::filesystem::exists(reboot_sentinel_);
+  bool need_reboot = false;
+
+  storage_.loadNeedReboot(&need_reboot);
+
+  return need_reboot && !beacon_exists;
+}
```

Comment:
The name here is slightly misleading, although given the context, it's probably fine. The problem is just that this does more than just detecting a reboot -- it's detecting an *expected* reboot. If we rebooted and weren't waiting for it, this would return false, despite that based on the name, you'd expect true. This isn't a real problem unless we reuse this logic for something else later without being careful.

Reply:
Yes, you're probably right. Do you have an alternate naming suggestion or should it just be better documented (well-placed comment?)

Reply:
No, I think it's fine for now. I can't think of anything better, so a good comment is probably enough.